rclpy/src/rclpy/impl/common.h

```diff
@@ -0,0 +1,125 @@
+// Copyright 2016 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#ifndef RCLPY__IMPL__COMMON_H_
+#define RCLPY__IMPL__COMMON_H_
+
+#include <Python.h>
+
+#include <rmw/types.h>
+
+typedef void * create_ros_message_signature (void);
+typedef void destroy_ros_message_signature (void *);
+typedef bool convert_from_py_signature (PyObject *, void *);
+typedef PyObject * convert_to_py_signature (void *);
+
+/// Convert a C rmw_qos_profile_t into a Python QoSProfile object
+/**
+ * \param[in] void pointer to a rmw_qos_profile_t structure
+ * \return QoSProfile object
+ */
+static PyObject *
+rclpy_convert_to_py_qos_policy(void * profile)
+{
+  PyObject * pyqos_module = PyImport_ImportModule("rclpy.qos");
+  if (!pyqos_module) {
+    return NULL;
+  }
+
+  PyObject * pyqos_policy_class = PyObject_GetAttrString(pyqos_module, "QoSProfile");
+  if (!pyqos_policy_class) {
+    return NULL;
+  }
+
+  PyObject * pyqos_profile = PyObject_CallObject(pyqos_policy_class, NULL);
+  Py_DECREF(pyqos_policy_class);
+  if (!pyqos_profile) {
+    return NULL;
```

Comment:
decref pyqos_module before returning, or do it once after getting `pyqos_policy_class` since it isn't used below

Reply:
Fixed 52afeca